tests/integration/test_storage_s3_queue/test_6.py

```diff
@@ -0,0 +1,281 @@
+import logging
+import time
+from multiprocessing.dummy import Pool
+
+import pytest
+
+from helpers.cluster import ClickHouseCluster
+from helpers.s3_queue_common import (
+    put_s3_file_content,
+    put_azure_file_content,
+    create_table,
+    create_mv,
+    generate_random_string,
+)
+
+
+@pytest.fixture(autouse=True)
+def s3_queue_setup_teardown(started_cluster):
+    instance = started_cluster.instances["instance"]
+    instance.query("DROP DATABASE IF EXISTS default; CREATE DATABASE default;")
+
+    minio = started_cluster.minio_client
+    objects = list(minio.list_objects(started_cluster.minio_bucket, recursive=True))
+    for obj in objects:
+        minio.remove_object(started_cluster.minio_bucket, obj.object_name)
+
+    container_client = started_cluster.blob_service_client.get_container_client(
+        started_cluster.azurite_container
+    )
+
+    if container_client.exists():
+        blob_names = [b.name for b in container_client.list_blobs()]
+        logging.debug(f"Deleting blobs: {blob_names}")
+        for b in blob_names:
+            container_client.delete_blob(b)
+
+    yield  # run test
+
+
+def put_file_content(started_cluster, engine_name, file_path, values_csv):
+    if engine_name == "S3Queue":
+        put_s3_file_content(started_cluster, file_path, values_csv)
+    else:
+        put_azure_file_content(started_cluster, file_path, values_csv)
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster = ClickHouseCluster(__file__)
+        cluster.add_instance(
```

Comment:
Let's also add a test with several clickhouse instances following the same keeper path, similar to `test_storage_s3_queue::test_2.py/test_shards_distributed`

Reply:
And let's add a test similar to `test_storage_s3_queue/test_5.py::test_failed_commit`